tests/mock_network/utils.rs

```diff
@@ -690,109 +715,20 @@ pub fn verify_section_invariants_for_node(node: &TestNode, elder_size: usize) {
     };
     if !all_neighbours_covered {
         panic!(
-            "{} Some neighbours aren't fully covered by the chain: {:?}",
-            node.name(),
+            "{}({:b}) Some neighbours aren't fully covered by the chain: {:?}",
+            our_name,
+            our_prefix,
             iter::once(*our_prefix)
                 .chain(neighbour_prefixes)
-                .collect::<Vec<_>>()
+                .format(", ")
         );
     }
 }
 
-pub fn verify_section_invariants_for_nodes(nodes: &[TestNode], elder_size: usize) {
-    for node in nodes.iter() {
-        verify_section_invariants_for_node(node, elder_size);
-    }
-}
-
-pub fn verify_section_invariants_between_nodes(nodes: &[TestNode]) {
-    #[derive(Debug)]
-    struct NodeSectionInfo {
-        node_name: XorName,
-        node_prefix: Prefix<XorName>,
-        view_section_version: u64,
-        view_section_elders: BTreeSet<XorName>,
-    };
-    let mut sections: BTreeMap<Prefix<XorName>, NodeSectionInfo> = BTreeMap::new();
-
-    for node in nodes.iter().filter(|node| node.inner.is_elder()) {
-        let our_prefix = node.our_prefix();
-        let our_name = node.name();
-        // NOTE: using neighbour_prefixes() here and not neighbour_infos().prefix().
-        // Is this a problem?
-        for prefix in iter::once(our_prefix).chain(node.inner.neighbour_prefixes().iter()) {
-            let our_info = NodeSectionInfo {
-                node_name: *our_name,
-                node_prefix: *our_prefix,
-                view_section_version: node.inner.section_elder_info_version(prefix).unwrap(),
-                view_section_elders: node.inner.section_elders(prefix),
-            };
-
-            if let Some(ref their_info) = sections.get(prefix) {
-                assert_eq!(
-                    (
-                        &our_info.view_section_elders,
-                        &our_info.view_section_version
-                    ),
-                    (
-                        &their_info.view_section_elders,
-                        &their_info.view_section_version
-                    ),
-                    "Section with prefix {:?} doesn't agree between nodes {:?} and \
-                     {:?}\n{:?},\n{:?}",
-                    prefix,
-                    our_info.node_name,
-                    their_info.node_name,
-                    our_info,
-                    their_info,
-                );
-                continue;
-            }
-            let _ = sections.insert(*prefix, our_info);
-        }
+pub fn verify_invariants_for_nodes(env: &Environment, nodes: &[TestNode]) {
+    for node in nodes {
+        verify_invariants_for_node(env, node);
     }
-
-    // check that prefixes are disjoint
```

Comment:
do we still need to check the prefixes are disjoint?

Reply:
No because they might not necessarily be at any given time.